File: leetcode/medium/1561. Maximum Number of Coins You Can Get/solution.cc
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

class Solution {
public:
    int maxCoins(vector<int>& piles) {
        int len = piles.size() * 2 / 3, sum =0;
        sort(piles.begin(), piles.end(), greater<>());

        for (int i = 1; i <= len; i += 2) {
            sum += piles[i];
        }
        return sum;
    }
};

int main() {
    Solution solution;
    vector<int> vec = {8, 7, 4, 2, 2, 1};
    cout << solution.maxCoins(vec);
    return 0;
}

File: leetcode/medium/1561. Maximum Number of Coins You Can Get/solution.js
// https://leetcode.com/problems/maximum-number-of-coins-you-can-get/
/**
 * @param {number[]} piles
 * @return {number}
 */
function maxCoins(piles) {
    const len = piles.length / 3 * 2;
    let sum = 0;
    piles.sort((a, b) => b - a);

    for (let i = 1; i <= len; i += 2) {
        sum += piles[i];
    }
    return sum;
};

// maxCoins([9, 8, 7, 6, 5, 1, 2, 3, 4])
maxCoins([2, 4, 1, 2, 7, 8])